nevergrad/optimization/test_optimizerlib.py

```diff
@@ -321,3 +321,12 @@ def test_instrumentation_optimizer_reproducibility() -> None:
     optimizer = optimizerlib.RandomSearch(instrumentation, budget=10)
     recom = optimizer.minimize(_square)
     np.testing.assert_equal(recom.kwargs["y"], 67)
+
+
+def test_constrained_optimization() -> None:
+    instrumentation = inst.Instrumentation(x=inst.var.Array(1), y=inst.var.Scalar())
+    optimizer = optimizerlib.OnePlusOne(instrumentation, budget=100)
```

Comment:
We're up to maaaaany more problems eventually with this constraints I think :D let's solve them when we find them.
First one: OnePlusOne always ask zero at the first iteration, so because num_ask did not change it would just always ask for zero with never satisfies the constraint. I made some hack to avoid this (temporarily updating num_ask) but it may not be very robust)

Reply:
I agree for one plus one, but I had solve this in a previous PR (by trying only a limited number of times), maybe just git rebase.

Reply:
For now I am sending an error when I cant bypass the error, I'll change to avoid this